### DISCUSSION TITLE
How to completely opt out of pre-rendering for route handlers

### DISCUSSION THREAD

**Discussion by davecarlson** [vercel/next.js#50656] [App Router]
I currently have the following in a route handler, which is a webhook handler for an api (there is no react, it takes in a body, does something, returns success/fail as json).

Under no circumstances do I want something like this being executed at bulid time, yet I can't see a 'good way' of stopping it, outside of moving it into the pages directly.

I've tried force dynamic and the other options (shown here), but when doing npm run build, it's still trying to execute the code.

Am I missing something obvious? 
it would be lovely to have like
```
export const preRendering = false
```
as an eject.

As an aside, there was nothing in the docs about how to parse form data, only json bodys. So I had to figure that out on my own.

```
async function handler(req, { params }) {
    const body: any = await req.formData();
// do my stuff
// result = { something: ""}
  return NextResponse.json({ result })
}
export { handler as GET, handler as POST }
export const fetchCache = 'force-no-store';
export const revalidate = 0
export const dynamic = 'force-dynamic'
```

**Comment by vimtor** [comment]
Oddly enough, this was causing my `send-email` cron job to be run during build time but after sending the email it stopped so I never registered the email as sent causing duplicated emails. I fixed it by checking for the `vercel-cron` user agent in the request headers.

**Comment by davecarlson** [comment]
That is definitely not the way we should have to do that ---_---

**Comment by baba43** [comment]
I have the same question and no idea why this is the default behaviour

**Comment by bestickley** [comment]
I'm using this:
```ts
/**
 * Check if code is running as a result of `next build`
 */
export function isNextBuild() {
  // eslint-disable-next-line turbo/no-undeclared-env-vars
  return process.env["NEXT_PHASE"] === "phase-production-build";
}
```
That's probably not a "public API" so could break in future but good enough for me.
